File: 0338.比特位计数/0338-比特位计数.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* countBits(int n, int* returnSize){
    int *res = (int *)malloc(sizeof(int) * (n + 1));
    res[0] = 0;
    for(int i = 1; i <= n; i++)
    {
        res[i] = res[i & (i - 1)] + 1;
    }

    *returnSize = n + 1;
    return res;
}

File: 0338.比特位计数/0338-比特位计数.java
class Solution {
    public int[] countBits(int n) {
        int res[] = new int[n + 1];//加 1
        res[0] = 0;
        for(int i = 1; i <= n; i++)//从 1 开始，到等于 n
        {
            res[i] = res[i & (i - 1)] + 1;
        }
        return res;
    }
}